department-of-veterans-affairs/va.gov-cms
# VACMS-6341: Top task content types.

## Description

Relates to #6341.

## Testing done

Behat, visual.

## Screenshots


## QA steps

For now, none of the editorial experience here should be visible to non-content-admin or admin users


## test content model


- [x] Set three Services taxonomy terms to "non-clinical services"

1. Go to /taxonomy/term/1026/edit, /taxonomy/term/1025/edit, and /taxonomy/term/1027/edit
2. You should be able to set these each to VAMC type of care: non-clinical service.

- [x] Make sure these are not exposed in current UI

3. Go to /node/add/regional_health_care_service_des ... make sure that terms 1025, 1026, and 1027 are not visible from the dropdown (they have been excluded via the change to the entity reference source view. 


- [ ] Test non-clincal health services

4. Go to /node/add/vha_facility_nonclinical_service, you should see these three taxonomy terms from the Service dropdown
5. You should be able to pick a facility, and a section. 
6. Go ahead and add three facility non-clinical service nodes, one for each of the 3 services. Assign to Aberdeen VA clinci and to Section: VA Sioux Falls (in VISN 23). 

- [ ] See that top task pages are now "service aware"

7. Create one of each of three top task pages (VAMC system Medical records, VAMC system Billing and insurance, and VAMC system Register for care), and again assign field_office and Section to VA Sioux Falls. 
8. Note that  `field_service_name_and_descri` has been set for each of these via their default field settings. if you want, you can unhide the field on the content type form, go back and edit the node, and see that it has been populated as expected with the relevant Service.  

## Test GraphQl feasibility

This should be enough data to generate a GraphQL query to use a node from any of the three content type content types to get: 

1. a VAMC system node ID via `field_office` (in this case, 8135 for VA Sioux Falls)
2. a single taxonomy term for `field_service_name_and_descri` (1025, 1026, or 1027)
3. an array of VAMC facility non-clinical service nodes associated to facilities that also have `field_region_page` set to field_office` and `field_service_name_and_descri` set to the taxonomy term. 



### Definition of Done


### Select Team for PR review


### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
